NPE: Could not pass event InventoryClickEvent to BentoBox v2.0.0-SNAPSHOT-b2514

### Expected behavior

Not sure

### Observed/Actual behavior

```
com.destroystokyo.paper.exception.ServerEventException: Could not pass event InventoryClickEvent to BentoBox v2.0.0-SNAPSHOT-b2514
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:72)
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126)
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615)
	at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:3117)
	at net.minecraft.network.protocol.game.PacketPlayInWindowClick.a(PacketPlayInWindowClick.java:58)
	at net.minecraft.network.protocol.game.PacketPlayInWindowClick.a(PacketPlayInWindowClick.java:23)
	at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$ensureRunningOnSameThread$0(PlayerConnectionUtils.java:54)
	at net.minecraft.server.TickTask.run(TickTask.java:18)
	at net.minecraft.util.thread.IAsyncTaskHandler.d(IAsyncTaskHandler.java:153)
	at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(IAsyncTaskHandlerReentrant.java:24)
	at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1458)
	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:194)
	at net.minecraft.util.thread.IAsyncTaskHandler.x(IAsyncTaskHandler.java:126)
	at net.minecraft.server.MinecraftServer.bl(MinecraftServer.java:1435)
	at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1358)
	at net.minecraft.util.thread.IAsyncTaskHandler.c(IAsyncTaskHandler.java:136)
	at net.minecraft.server.MinecraftServer.w_(MinecraftServer.java:1336)
	at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1220)
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:321)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.NullPointerException: Cannot invoke "Object.getClass()" because "key" is null
	at java.base/java.util.EnumMap.typeCheck(EnumMap.java:736)
	at java.base/java.util.EnumMap.put(EnumMap.java:264)
	at java.base/java.util.EnumMap.put(EnumMap.java:78)
	at java.base/java.util.AbstractMap.putAll(AbstractMap.java:281)
	at java.base/java.util.EnumMap.putAll(EnumMap.java:347)
	at Level-2.12.0-SNAPSHOT-b553.jar//world.bentobox.level.panels.DetailsPanel.updateFilters(DetailsPanel.java:117)
	at Level-2.12.0-SNAPSHOT-b553.jar//world.bentobox.level.panels.DetailsPanel.<init>(DetailsPanel.java:66)
	at Level-2.12.0-SNAPSHOT-b553.jar//world.bentobox.level.panels.DetailsPanel.openPanel(DetailsPanel.java:592)
	at Level-2.12.0-SNAPSHOT-b553.jar//world.bentobox.level.panels.TopLevelPanel.lambda$createIslandIcon$2(TopLevelPanel.java:223)
	at BentoBox.jar//world.bentobox.bentobox.listeners.PanelListenerManager.lambda$onInventoryClick$0(PanelListenerManager.java:56)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at BentoBox.jar//world.bentobox.bentobox.listeners.PanelListenerManager.onInventoryClick(PanelListenerManager.java:54)
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor964.execute(Unknown Source)
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77)
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:77)
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54)
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126)
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615)
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleContainerClick(ServerGamePacketListenerImpl.java:3117)
	at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:58)
	at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:23)
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:54)
	at net.minecraft.server.TickTask.run(TickTask.java:18)
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153)
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24)
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1458)
	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:194)
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126)
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1435)
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1358)
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136)
	at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1336)
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1220)
	... 2 more
```

### Steps/models to reproduce

Unable to reproduce yet.

### BentoBox version

 bentobox version
[19:39:06 INFO]: Running PAPER 1.20.4.
[19:39:06 INFO]: BentoBox version: 2.0.0-SNAPSHOT-b2514
[19:39:06 INFO]: Database: JSON
[19:39:06 INFO]: Loaded Game Worlds:
[19:39:06 INFO]: bskyblock_world (BSkyBlock): Overworld, Nether, The End
[19:39:06 INFO]: Loaded Addons:
[19:39:06 INFO]: Bank 1.7.1-SNAPSHOT-b87 (ENABLED)
[19:39:06 INFO]: Biomes 2.2.0-SNAPSHOT-b255 (ENABLED)
[19:39:06 INFO]: Border 4.1.1-SNAPSHOT-b187 (ENABLED)
[19:39:06 INFO]: BSkyBlock 1.17.0-SNAPSHOT-b751 (ENABLED)
[19:39:06 INFO]: CauldronWitchery 2.1.0-SNAPSHOT-#58 (ENABLED)
[19:39:06 INFO]: Challenges 1.3.0-SNAPSHOT-b538 (ENABLED)
[19:39:06 INFO]: CheckMeOut 1.2.0-SNAPSHOT-b92 (ENABLED)
[19:39:06 INFO]: ControlPanel 1.13.1-SNAPSHOT-b61 (ENABLED)
[19:39:06 INFO]: DimensionalTrees 1.6.1-SNAPSHOT-b89 (ENABLED)
[19:39:06 INFO]: ExtraMobs 1.12 (ENABLED)
[19:39:06 INFO]: IslandFly 1.11.1 (ENABLED)
[19:39:06 INFO]: Level 2.12.0-SNAPSHOT-b553 (ENABLED)
[19:39:06 INFO]: Likes 2.4.0-SNAPSHOT-b99 (ENABLED)
[19:39:06 INFO]: MagicCobblestoneGenerator 2.6.0-SNAPSHOT-b270 (ENABLED)
[19:39:06 INFO]: TwerkingForTrees 1.4.4-SNAPSHOT-b68 (ENABLED)
[19:39:06 INFO]: Visit 1.7.0-SNAPSHOT (ENABLED)
[19:39:06 INFO]: Warps 1.14.0-SNAPSHOT-b358 (ENABLED)


### Plugin list

_No response_

### Other

_No response_

--- a/pom.xml
+++ b/pom.xml
@@ -192,6 +192,31 @@
     </repositories>
 
     <dependencies>
+        <!-- Mockito (Unit testing) This goes at the top to ensure the dependencies are accurate. -->
+        <!-- This is required for PowerMockito to work and must be placed before it -->
+        <dependency>
+            <groupId>org.javassist</groupId>
+            <artifactId>javassist</artifactId>
+            <version>3.30.2-GA</version>
+       </dependency>
+       <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-module-junit4</artifactId>
+            <version>${powermock.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-api-mockito2</artifactId>
+            <version>${powermock.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>3.11.1</version>
+            <scope>test</scope>
+        </dependency>
         <!-- Spigot API -->
         <dependency>
             <groupId>org.spigotmc</groupId>
@@ -211,25 +236,6 @@
             <groupId>org.bstats</groupId>
             <artifactId>bstats-bukkit</artifactId>
             <version>${bstats.version}</version>
-        </dependency>
-        <!-- Mockito (Unit testing) -->
-        <dependency>
-            <groupId>org.mockito</groupId>
-            <artifactId>mockito-core</artifactId>
-            <version>3.11.1</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.powermock</groupId>
-            <artifactId>powermock-module-junit4</artifactId>
-            <version>${powermock.version}</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.powermock</groupId>
-            <artifactId>powermock-api-mockito2</artifactId>
-            <version>${powermock.version}</version>
-            <scope>test</scope>
         </dependency>
         <!-- Database -->
         <dependency>

--- a/src/main/java/world/bentobox/bentobox/database/json/BentoboxTypeAdapterFactory.java
+++ b/src/main/java/world/bentobox/bentobox/database/json/BentoboxTypeAdapterFactory.java
@@ -3,6 +3,7 @@ package world.bentobox.bentobox.database.json;
 import java.util.Map;
 
 import org.bukkit.Location;
+import org.bukkit.Material;
 import org.bukkit.World;
 import org.bukkit.block.Biome;
 import org.bukkit.configuration.serialization.ConfigurationSerializable;
@@ -23,6 +24,7 @@ import world.bentobox.bentobox.database.json.adapters.EnumTypeAdapter;
 import world.bentobox.bentobox.database.json.adapters.FlagTypeAdapter;
 import world.bentobox.bentobox.database.json.adapters.ItemStackTypeAdapter;
 import world.bentobox.bentobox.database.json.adapters.LocationTypeAdapter;
+import world.bentobox.bentobox.database.json.adapters.MaterialTypeAdapter;
 import world.bentobox.bentobox.database.json.adapters.PotionEffectTypeAdapter;
 import world.bentobox.bentobox.database.json.adapters.VectorTypeAdapter;
 import world.bentobox.bentobox.database.json.adapters.WorldTypeAdapter;
@@ -55,6 +57,8 @@ public class BentoboxTypeAdapterFactory implements TypeAdapterFactory {
         if (Location.class.isAssignableFrom(rawType)) {
             // Use our current location adapter for backward compatibility
             return (TypeAdapter<T>) new LocationTypeAdapter();
+        } else if (Material.class.isAssignableFrom(rawType)) {
+            return (TypeAdapter<T>) new MaterialTypeAdapter();
         } else if (Biome.class.isAssignableFrom(rawType)) {
             return (TypeAdapter<T>) new BiomeTypeAdapter();
         } else if (Enum.class.isAssignableFrom(rawType)) {

--- a/src/main/java/world/bentobox/bentobox/database/json/adapters/MaterialTypeAdapter.java
+++ b/src/main/java/world/bentobox/bentobox/database/json/adapters/MaterialTypeAdapter.java
@@ -1,0 +1,56 @@
+package world.bentobox.bentobox.database.json.adapters;
+
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.bukkit.Material;
+
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+
+
+/**
+ * Minecraft 1.20 changed GRASS to SHORT_GRASS. This class provides and backwards compatibility when loading
+ * databased files stored with previous versions. It can be extended in the future if further enum changes are made.
+ * @author tastybento
+ * @since 2.0.0
+ */
+public final class MaterialTypeAdapter extends TypeAdapter<Material>
+{
+    /**
+     * Map that contains string value to the actual Material enum object.
+     */
+    final Map<String, Material> materialMap;
+
+    public MaterialTypeAdapter() {
+        this.materialMap = new HashMap<>();
+
+        // Put in current values.
+        Arrays.stream(Material.values()).forEach(mat -> this.materialMap.put(mat.name(), mat));
+
+        // Put in renamed material values.
+        this.materialMap.put("GRASS", Material.SHORT_GRASS);
+    }
+
+    @Override
+    public Material read(JsonReader input) throws IOException
+    {
+        if (JsonToken.NULL.equals(input.peek())) {
+            input.nextNull();
+            return null;
+        }
+
+        return this.materialMap.get(input.nextString().toUpperCase());
+    }
+
+    @Override
+    public void write(JsonWriter output, Material enumValue) throws IOException {
+        output.value(enumValue != null ? enumValue.name() : null);
+    }
+}
+

